meilisearch/client.py

```diff
@@ -80,6 +81,34 @@ def get_index(self, uid):
         """
         return Index.get_index(self.config, uid=uid)
 
+    def get_or_create_index(self, uid, options=None):
+        """Get an index, or create it if it doesn't exist.
+
+        Parameters
+        ----------
+        uid: str
+            UID of the index
+        options: dict, optional
+            Options passed during index creation (ex: primaryKey)
+
+        Returns
+        -------
+        index : Index
+            an instance of Index containing the information of the retrieved or newly created index
+        Raises
+        ------
+        HTTPError
+            In case of any other error found here https://docs.meilisearch.com/references/#errors-status-code
+        """
+        try:
+            index = Index(self.config, uid)
+            index.create(self.config, uid, options)
+        except MeiliSearchApiError as err:
+            if err.message != 'index already exists':
```

Comment:
Why don't you use the `code` attributes that MeiliSearch finally (🙏) integrates into the errors in the v0.11? 🙃

Reply:
I will do this after we merge #117, thanks :)